### PR TITLE
Do not delete push task names immediately

### LINKED ISSUE #2762: Task names are not tombstoned after completion
When tasks are expired, failed, or completed, the task name is made immediately available again rather than kept as a marker to prevent other tasks with the same name from being enqueued for awhile.

### PR DESCRIPTION
**Pull request by cdonati**
This allows task names to be tombstoned for awhile like they are in GAE. It prevents clients from re-enqueuing the same task multiple times.

The TaskName entities are cleaned up by the groomer after a day.

Resolves #2762.

### CODE REVIEW

File: AppTaskQueue/appscale/taskqueue/distributed_tq.py [hunk 1]
```diff
@@ -574,19 +557,23 @@ def __check_and_store_task_names(self, request):
...
-      except datastore_errors.InternalError, internal_error:
-        logger.error(str(internal_error))
+      if item.state == TASK_STATES.QUEUED:
+        logger.warning("Task already exists")
         raise apiproxy_errors.ApplicationError(
-          TaskQueueServiceError.DATASTORE_ERROR)
+          TaskQueueServiceError.TASK_ALREADY_EXISTS)
+      else:
+        raise apiproxy_errors.ApplicationError(
+          TaskQueueServiceError.INVALID_TASK_NAME)
```
Comment (jeanleonov) [inline AppTaskQueue/appscale/taskqueue/distributed_tq.py]: Does GAE somehow differentiate invalid task name like this case (used name) and invalid name containing some special symbols? Is it easy to add some descriptive message here?

Comment (jeanleonov) [reply AppTaskQueue/appscale/taskqueue/distributed_tq.py]: I don't take my approval)

Comment (cdonati) [reply AppTaskQueue/appscale/taskqueue/distributed_tq.py]: Good catch. I thought when I tested earlier that GAE used InvalidTaskNameError for tombstoned task names, but when I just tested now, it returned TombstonedTaskError.

**Comment by cdonati** [comment]
@jeanleonov Yes, this was the tombstoning behavior I mentioned in your PR. Your plan to keep the Postgres PR as it currently is sounds good to me.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/5271/
https://ocd.appscale.com:8080/job/Tools-test/3537/

**Comment by cdonati** [comment]
Demo
https://github.com/AppScale/hawkeye/pull/139 with master: https://ocd.appscale.com:8080/job/Hawkeye/15324/
https://github.com/AppScale/hawkeye/pull/139 with this branch: https://ocd.appscale.com:8080/job/Hawkeye/15323/

**Pull request merged**